File: frontend/src/types/entity/Host.ts
export interface HostGroup {
  id: number;
  projectId: number;
  name: string;
  description: string;
  createdBy: number;
  createdAt: string;
}

export interface Host {
  id: number;
  hostGroupId: number;
  name: string;
  ip: string;
  port: number;
  ansibleUser: string;
  ansibleSshPass: string;
  ansibleSshPrivateKeyFile: string;
  ansibleBecome: boolean;
  createdBy: number;
  createdAt: string;
}

export interface CreateHostGroupRequest {
  name: string;
  description?: string;
}

export interface UpdateHostGroupRequest {
  name?: string;
  description?: string;
}

export interface CreateHostRequest {
  name: string;
  ip: string;
  port?: number;
  ansibleUser?: string;
  ansibleSshPass?: string;
  ansibleSshPrivateKeyFile?: string;
  ansibleBecome?: boolean;
}

export interface UpdateHostRequest {
  name?: string;
  ip?: string;
  port?: number;
  ansibleUser?: string;
  ansibleSshPass?: string;
  ansibleSshPrivateKeyFile?: string;
  ansibleBecome?: boolean;
}

File: backend/src/main/java/com/ansible/host/entity/Host.java
package com.ansible.host.entity;

import com.ansible.common.BaseEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Index;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(
    name = "hosts",
    indexes = {@Index(name = "idx_host_host_group_id", columnList = "host_group_id")})
@Getter
@Setter
@NoArgsConstructor
public class Host extends BaseEntity {

  @Column(nullable = false)
  private Long hostGroupId;

  @Column(nullable = false, length = 100)
  private String name;

  @Column(nullable = false, length = 45)
  private String ip;

  @Column(nullable = false)
  private Integer port = 22;

  @Column(length = 100)
  private String ansibleUser;

  @Column(length = 500)
  private String ansibleSshPass;

  @Column(length = 2000)
  private String ansibleSshPrivateKeyFile;

  @Column(nullable = false)
  private Boolean ansibleBecome = false;
}